packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTableRowAction.tsx

```diff
@@ -76,13 +53,32 @@ export function TransactionsTableRowAction({
     return isClaiming || isClaimingCctp || !isConfirmed
   }, [isClaiming, isClaimingCctp, isConfirmed])
 
-  const isRedeemButtonDisabled = useMemo(
-    () =>
-      typeof isConnectedToArbitrum !== 'undefined'
-        ? !isConnectedToArbitrum
-        : true,
-    [isConnectedToArbitrum]
-  )
+  const isConnectedToCorrectNetworkForRedeem = useMemo(() => {
+    if (!chain) {
+      return false
+    }
+    return chain.id === tx.childChainId
+  }, [chain, tx.childChainId])
+
+  const handleRedeemRetryable = useCallback(async () => {
+    try {
+      if (!isConnectedToCorrectNetworkForRedeem) {
+        await switchNetworkAsync?.(tx.childChainId)
+      }
+      redeem()
+    } catch (error: any) {
+      if (isUserRejectedError(error)) {
+        return
+      }
+
+      errorToast(`Can't retry the deposit: ${error?.message ?? error}`)
```

Comment:
We need to reset `isRedeeming` to false in a finally block because when I reject redeem, it still shows retrying.

<img width="400" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/bacdd2e2-3b6c-4f3c-8c9d-f63c3e2d7ce3">

Reply:
I've removed the finally block based on the refactor where we had 2 try catch blocks, should work fine now